pkg/wasm/convert.go

```diff
@@ -74,7 +74,7 @@ func MaybeConvertWasmExtensionConfig(resources []*anypb.Any, cache Cache) bool {
 func convert(resource *anypb.Any, cache Cache) (newExtensionConfig *anypb.Any, sendNack bool) {
 	ec := &core.TypedExtensionConfig{}
 	newExtensionConfig = resource
-	sendNack = false
+	sendNack = true
```

Comment:
I feel fairly uncomfortable with this.. a NACK is a *huge* risk, has the possibility to cause cluster wide outage by blocking all pod startup. Its worth a `return nil, true` to explicitly signal this rather than implicitly NACKing

Reply:
Return like this is bad practice in Go in general, IMO, but especially in this case

Reply:
Actually i guess in this case since we are proxying its a bit different but still a good policy. 

Also its not clear - can `convert` return `nil, false`? What happens if so?

Also - On NACK, shouldn't we apply the valid configs and just NACK the invalid ones?

Reply:
The problem is that a valid xDS causes Envoy crash. The only mechanism to prevent it is NACK in the proxy. A crash is worse than inability to update some (optional) config.

Reply:
I am not saying we shouldn't NACK, just that the default being NACK and then unsettting it is confusing/risky. Just a syntax concern; we should explicitly annotate on each return nack or not

Reply:
Right, that's why I asked for the coverage report. We should have unit tests for good/bad cases and make sure they are reasonable.

Reply:
@kyessenov Here is the coverage report. 
https://storage.googleapis.com/igsong-test-wasms/coverage/istio-wasm-coverage.html#file1